lib/swagger/formatter.ex

```diff
@@ -134,46 +134,68 @@ defmodule Xcribe.Swagger.Formatter do
   @doc """
   Merge two lists of parameter object keep uniq names
   """
-  def merge_parameter_object_lists(base_list, new_list) do
-    Enum.reduce(new_list, base_list, &merge_parameter_func/2)
+  def merge_parameter_object_lists(base_list, new_list, mode \\ :keep) do
+    new_list
+    |> Enum.reduce(base_list, &merge_parameter_func(&1, &2, mode))
+    |> Enum.sort(&(&1.name < &2.name))
   end
 
   @doc """
   Merge two path item objects
   """
-  def merge_path_item_objects(item_one, item_two, verb) do
+  def merge_path_item_objects(base, new_item, verb) do
     Map.update(
-      item_one,
+      base,
       verb,
-      item_two[verb],
-      &merge_path_items(&1, item_two[verb])
+      new_item[verb],
+      &merge_path_items(&1, new_item[verb])
     )
   end
 
-  defp merge_path_items(item_one, %{parameters: params, responses: resp} = item_two) do
-    item_one
-    |> Map.update(:parameters, params, &merge_parameter_object_lists(&1, params))
+  defp merge_path_items(base, %{parameters: params, responses: resp} = new_item) do
+    mode = overwrite_mode(resp)
+
+    base
+    |> Map.update(:parameters, params, &merge_parameter_object_lists(&1, params, mode))
     |> Map.update(:responses, resp, &Map.merge(&1, resp))
-    |> merge_request_body_if_needed(item_two)
+    |> merge_request_body_if_needed(new_item, mode)
   end
 
-  defp merge_request_body_if_needed(%{requestBody: _} = item, %{requestBody: new_body}) do
-    Map.update(item, :requestBody, new_body, &merge_request_body(&1, new_body))
+  defp overwrite_mode(responses) do
+    code = responses |> Map.keys() |> List.first()
```

Comment:
Is `code` always the first Map key of Response? Is there any other way to make it clearer that we're aiming at that particular key? :thinking:

Reply:
The struct is like that:

```elixir
      expected = %{
        "get" => %{
          description: "",
          parameters: [],
          security: [%{"api_key" => []}],
          summary: "",
          tags: ["users"],
          responses: %{
            200 => %{description: "", headers: , content: %{  "application/json" => %{ schema: } } }
          }
        }
      }
```
The response map have the requests status code as keys. As we generate one path at a time the map has just one key. I don't know what key is coming, but I know it is just one status code.